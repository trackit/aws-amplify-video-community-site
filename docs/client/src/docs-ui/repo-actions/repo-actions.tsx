import {Component, h, Host, Prop} from "@stencil/core";
import {repoActionsStyle} from "./repo-actions.style";
import {Page} from "../../api";
import {createIssueLink} from "../../utils/issue-link";

@Component({tag: "docs-repo-actions", shadow: false})
export class DocsRepoActions {
  /*** the page from which a subpath can be attained */
  @Prop() readonly page?: Page;

  render() {
    return (
      <Host class={repoActionsStyle}>
        <amplify-external-link href={createIssueLink()}>
          <img src="/assets/flag.svg" alt="Feedback" />
          Feedback
        </amplify-external-link>
        <amplify-external-link
          href={`https://github.com/aws-amplify/docs/edit/main/docs/${this.page?.relativeToContentDir}`}
        >
          <img src="/assets/github.svg" alt="Edit" />
          Edit
        </amplify-external-link>
      </Host>
    );
  }
}
