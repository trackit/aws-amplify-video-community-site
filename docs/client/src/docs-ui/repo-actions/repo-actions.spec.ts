import {DocsRepoActions} from "./repo-actions";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-repo-actions", () => {
  describe("Component logic", () => {
    let repoActions: DocsRepoActions;
    beforeEach(() => (repoActions = new DocsRepoActions()));

    it("should init `page` as `undefined`", () =>
      expect(repoActions.page).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsRepoActions],
            html: `<docs-repo-actions />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
