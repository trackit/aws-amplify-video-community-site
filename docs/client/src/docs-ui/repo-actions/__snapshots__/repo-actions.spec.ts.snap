// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-repo-actions Render logic should render 1`] = `
<docs-repo-actions class="css-1bbh6oj">
  <amplify-external-link href="https://github.com/aws-amplify/docs/issues/new?title=[Feedback]FEEDBACK_TITLE_HERE&amp;labels=&amp;body=**Page**%3A%20%5B%60%2F%60%5D(http%3A%2F%2Ftesting.stenciljs.com%2F)%0A%0A**Feedback**%3A%0A%0A%3C!--%20your%20feedback%20here%20--%3E">
    <img alt="Feedback" src="/assets/flag.svg">
    Feedback
  </amplify-external-link>
  <amplify-external-link href="https://github.com/aws-amplify/docs/edit/main/docs/undefined">
    <img alt="Edit" src="/assets/github.svg">
    Edit
  </amplify-external-link>
</docs-repo-actions>
`;
